controllers/limitador_controller.go

```diff
@@ -78,7 +79,27 @@ func (r *LimitadorReconciler) Reconcile(eventCtx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
-	deployment := limitador.LimitadorDeployment(limitadorObj)
+	limitadorStorage := limitadorObj.Spec.Storage
+	storageConfigSecret := &v1.Secret{}
+	if limitadorStorage != nil && limitadorStorage.Type != limitadorv1alpha1.StorageTypeInMemory {
+		if limitadorStorage.ConfigSecretRef == nil {
+			logger.Error(err, "There's no ConfigSecretRef set.")
```

Comment:
this is a kind of validation types that can be done in a method 
```go
func (l *Limitador) Validate() error {}
```

And call it before any reconciliation.